kafka/consumer.go

```diff
@@ -342,10 +332,17 @@ func (c *KafkaConsumer) Close() error {
 	if err := c.client.Close(); err != nil {
 		errorStrs = append(errorStrs, err.Error())
 	}
-	close(c.errors)
-	// drains all errors and return them.
-	for errs := range c.errors {
-		errorStrs = append(errorStrs, errs.Error())
+
+	// drains all errors and return them - don't close errors chan because this causes a panic when goroutines try to send on it
+	for {
+		select {
+		case err := <-c.errors:
+			errorStrs = append(errorStrs, err.Error())
+			continue
+		default:
```

Comment:
^ but that expects the errors chan to close before it exits the loop. i was trying to come up with a way to drain the channel without closing it because that causes a panic sometimes

Reply:
I made a change to only send to the errors chan from the `constructEvent` method, so maybe i can just close the chan there and revert this back to what you had

Reply:
👍